pkg/cliplugins/workspace/cmd/cmd.go

```diff
@@ -204,6 +209,49 @@ func NewCmdWorkspace(streams genericclioptions.IOStreams) (*cobra.Command, error
 	}
 	createContextCmd.Flags().BoolVar(&overwriteContext, "overwrite", overwriteContext, "Overwrite the context if it already exists")
 
+	// Any user-specified types will be synced in addition to the required types
+	// to ensure support for the use case of a synced deployment capable of
+	// talking to kcp.
+	//
+	// TODO(marun) Consider allowing a user-specified and exclusive set of types.
+	requiredResourcesToSync := sets.NewString("deployments.apps", "secrets", "configmaps", "serviceaccounts")
+
+	var resourcesToSync []string
+	var syncerImage string
+	disableDeployment := false
+	enableSyncerCmd := &cobra.Command{
+		Use:          "enable-syncer <workload-cluster-name> [--sync-resources=<resource1>,<resource2>..]",
```

Comment:
is this designed to be idem-potent?

Reply:
No, it's not. The limiting factor is the workload cluster name - if it already exists the command will fail without doing anything. If it does exist (and e.g. a previous command didn't complete successfully), deleting the workload cluster ensures removal of all related resources due to owner references.